### PR TITLE
Honor `getValue` method of ReflectionToStringBuilder & its subclasses to get the field value

### PR DESCRIPTION
**Pull request by shashankrnr32**
Noticed this change recently which removed the usage of the method `getValue` to get the field value. Custom implementations of the `getValue` would not be used anymore.

@garydgregory

### CODE REVIEW

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @shashankrnr32 
Thank you for your PR!
This PR missing a unit test to avoid a future regression.

File: src/main/java/org/apache/commons/lang3/builder/ReflectionToStringBuilder.java [hunk 1]
```diff
@@ -703,13 +703,11 @@ public Class<?> getUpToClass() {
      *
      * @throws IllegalArgumentException
      *             see {@link java.lang.reflect.Field#get(Object)}
-     * @throws IllegalAccessException
-     *             see {@link java.lang.reflect.Field#get(Object)}
      *
      * @see java.lang.reflect.Field#get(Object)
      */
-    protected Object getValue(final Field field) throws IllegalAccessException {
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/lang3/builder/ReflectionToStringBuilder.java:711]: Why are you removing `IllegalAccessException`?

**Comment by shashankrnr32** [comment]
> Hi @shashankrnr32 Thank you for your PR! This PR missing a unit test to avoid a future regression.

@garydgregory Added `ReflectionToStringBuilderCustomImplementationTest` that overrides the `getValue` and modifies the generated field value by the base class. If this behaviour changes in the future, the test will break.

**Comment by garydgregory** [comment]
https://issues.apache.org/jira/projects/LANG/issues/LANG-1710

File: src/main/java/org/apache/commons/lang3/builder/ReflectionToStringBuilder.java (hunk 1 above)
Comment (garydgregory) [reply src/main/java/org/apache/commons/lang3/builder/ReflectionToStringBuilder.java:711]: Hello @shashankrnr32 
Thank you for your update.
This PR attempts to restore compatibility with the `getValue()` method but changes that method's definition and breaks compatibility differently. For example, if I have a subclass that overrides `getValue()` using the released signature, that subclass will no longer compile if this patch is applied. The test in this PR does not reflect that a subclass can throw `IllegalAccessException` in the `getValue()` method signature, so the test is not checking for full compatibility. See git master for a fix.

**Comment by garydgregory** [comment]
Closing, see the comment.

**Pull request closed without merging**